lib/Doctrine/Annotations/Type/MixedType.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Annotations\Type;
+
+use function is_resource;
+
+/**
+ * @internal
+ */
+final class MixedType implements Type
+{
+    public function describe() : string
+    {
+        return 'mixed';
+    }
+
+    /**
+     * @param mixed $value
+     */
+    public function validate($value) : bool
+    {
+        return ! is_resource($value);
```

Comment:
How would you define a resource value in phpDoc? :)

Reply:
Exactly. So this isn't MixedType. This should just return true, since it's not possible to define it in phpdoc anyway.

Reply:
It is in context of annotations: resource is not a valid type.

Reply:
Don't leak assumptions over whole codebase. This file shouldn't know what is and what isn't possible to define in docblock. It says MixedType but doesn't accept resource. It's a problem. Fix naming or make it accept anything. Shouldn't be a big deal since you can't normally define resource in annotations as you say.

Reply:
It's part of Annotations library and it's not public API, it's fine to know overall context.

Reply:
So why is it caring about resource when this is annotations library?